File: ch5/5-3.c
#include <stdio.h>

void p_strcat(char *s, char *t);

int main(void)
{
    char name[100] = "Jonathan ";
    char *last_name = "Torres";
    p_strcat(name, last_name);
    printf("%s\n", name);
    return 0;
}

void p_strcat(char *s, char *t)
{
    // find end of s
    while (*s != '\0') {
        s++;
    }
    while ((*s = *t) != '\0') {
        // copy t
        s++;
        t++;
    }
}

File: gkr/ch5/5-3.go
package main

import "fmt"

func main() {
	name := make([]byte, 0, 100)
	name = append(name, []byte("Jonathan ")...)
	lastName := []byte("Torres")
	p_strcat(&name, &lastName)
	fmt.Printf("%s\n", string(name))
}

func p_strcat(s *[]byte, t *[]byte) {
	var i int
	// find the end of s
	for {
		if i >= len(*s)-1 {
			break
		}
		i++
	}
	for _, x := range *t {
		*s = append(*s, x)
	}
}
